src/ICSharpCode.SharpZipLib/Zip/ZipOutputStream.cs

```diff
@@ -313,6 +313,8 @@ internal void PutNextEntry(Stream stream, ZipEntry entry, long streamOffset = 0)
 				throw new InvalidOperationException("The Password property must be set before AES encrypted entries can be added");
 			}
 
+			entryIsPrecompressed = string.IsNullOrEmpty(Password) && method == CompressionMethod.Deflated && entry.Size >= 0 && entry.HasCrc && entry.CompressedSize >= 0;
```

Comment:
Is there any way user can know compressed size and crc? Compressed size could be changed because of compression level or simply compressor logic change. So i think no one will pass it and expect that it will match internal deflator deflated size. Except if user wants to pass its own data

Reply:
> So i think no one will pass it and expect

We can't consider this a non-breaking change just because it would seem illogical to use it in this way.
It's still the fact that code that put something in those properties for any reason (copying an entry object from another file, serialization, left-over code?) would suddenly stop working. At least it would throw an exception, but it's change of behavior which would warrant a new major release if done this way.

Reply:
It is already broken because any fixes in compression engine will make the code fail as compressed size will differ.
Anyway, i got the point and reworked the PR